### ISSUE TITLE
Consider stripping out only IMAGES as option

### ISSUE THREAD

**Issue by rsvp** [kynan/nbstripout#58]
Frequently the output is useful as informal "testing" of results, 
and there is very little overhead in keeping non-graphical results. 
***Images, however, add considerable unwanted bulk to a commit*** 
for any version control system (unless those images are very 
expensive to reproduce, or are historical for some reason).

Proposal: provide an **option to strip out only output cells with images**.

**Comment by kynan** [comment]
Interesting, that sounds like a useful option. How do you see the "UI" of this option?

Suppose this could even be easily generalized to support any data type to either keep or strip out.

**Comment by rsvp** [comment]
As an example, how about along these lines?

```
$ nbstripout --only=images [notebook(s)]
#  where images is just an alias for *.png, *.jpg, etc.

$ nbstripout --only='tmp-*.png' [notebook(s)]
#  for inconsequential images labeled as such by user.
```

**Comment by jpeacock29** [comment]
I thought about this as I was implementing `--keep-count` and `--keep-output` and considered an options structure:

    --keep-count, -c
    --keep-output, -o
        --keep-text-output, -t
        --keep-image-output, -i
    --keep-metadata, -m

Thus `nbstripout -o` would be equivalent to `nbstripout -ti`. Stripping only images would use `nbstripout -ctm`. For more fine-grained control at the cell level, `keep_output`-type metadata would suffice. That said, this doesn't scale well as maybe there are more output types we'd consider in the future. (Widgets?)

**Comment by rsvp** [comment]
@jpeacock29 Re: more output types. Consider a flag like `--regex, -r`

where a user can implement their custom stripout by giving a regular expression.
This is what I did with `sed` -- then understood that an edited notebook
must leave in a **trusted state**. Going from regex to images would 
then be a one-liner.

**Comment by kynan** [comment]
Sounds quite sensible! Let's start with images since this seems to be the most important case.

One of you happy to have a go at this @rsvp @jpeacock29 ?

**Comment by jpeacock29** [comment]
What would the regex flag be matched against? Every key in the ipynb?

**Comment by rsvp** [comment]
@jpeacock29 here's a mock example for PNG images: `$ nbstripout --regex='png": "i'`

That *should* get rid of the super long lines encoding PNG images for now.
I question the permanence of that regular expression since it is subject
to the formatting whims upstream. Historically, both `"image/png":`
and `"png":` have been used as keys, so my example would fortunately 
work for both cases. 

But --regex would be a handy tool in any case.

**Comment by kynan** [comment]
Feel free to have a go at this, I'm not sure when I might find time to work on it myself.

**Comment by kynan** [comment]
One of you interested in working on this @rsvp @jpeacock29 ?

**Comment by kynan** [comment]
Are you still interested in this @rsvp @jpeacock29 ?

**Comment by rsvp** [comment]
hi @kynan my spare cycles are going to refactoring https://git.io/fecon235
so realistically *maybe* later this year.

Interestingly, one of the reasons leading to the spin-off of the source code
to another repository https://git.io/fecon236 was to leave behind all
the archival *bulky* images preserved in the .git for notebooks. 

So this issue is still pertinent.

**Comment by kynan** [comment]
@rsvp do you still have this use case?

**Comment by kynan** [comment]
There's an in flight pull request (#135) that's somewhat related: only strip outputs that are larger than a certain size. Would that fit the bill?

**Comment by kynan** [comment]
Given #135 has been released in nbstripout 0.5.0 and is arguably even more flexible than what's requested here I'll close this as fixed.
